devdoc/constbuffer_requirements.md

```diff
@@ -28,13 +27,13 @@ typedef struct CONSTBUFFER_HANDLE_DATA_TAG* CONSTBUFFER_HANDLE;
 typedef struct CONSTBUFFER_TAG
 {
     const unsigned char* buffer;
-    size_t size;
+    uint32_t size;
 } CONSTBUFFER;
 
 typedef void(*CONSTBUFFER_CUSTOM_FREE_FUNC)(void* context);
 
 /*what function should CONSTBUFFER_HANDLE_to_buffer use to allocate the returned serialized form. NULL means malloc from gballoc_hl_malloc_redirect.h*/
-typedef void*(*CONSTBUFFER_to_buffer_alloc)(size_t size);
+typedef void*(*CONSTBUFFER_to_buffer_alloc)(uint32_t size);
```

Comment:
>uint32_t [](http://example.com/codeflow?start=44&length=8)

I do not think this should be uint32_t.
The alloc functions do not use uint32_t, but size_t, so we should definitely keep this one in that same class.

While doing my own attack at this I realized that maybe one day we want to contain the seepage of size_t in the code due to alloc functions by having our wrappers be fixed size type based and containing the casts in our implementations of the alloc functions.

But frankly that should be a task for another day, otherwise we'll never merge something.